### PR TITLE
feat(ui): remove sidebar for node edit and hook edit

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-ui#9343.0 ✘  
* Compile
  * Build UI ✘  
  * TU All -   
  * TU model ✘  
  * TU service ✔ 
  * TU shared ✘  
  * TU store ✘  
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report build-ui#9348.0 ✘  
* Compile
  * Build UI ✘  
  * TU All -   
  * TU model ✘  
  * TU service ✔ 
  * TU shared ✘  
  * TU store ✘  
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report build-ui#9406.0 ✘  
* Compile
  * Build UI ✘  
  * TU All -   
  * TU model ✘  
  * TU service ✔ 
  * TU shared ✘  
  * TU store ✘  
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report build-ui#9413.0 ✘  
* Compile
  * Build UI ✘  
  * TU All -   
  * TU model ✘  
  * TU service ✔ 
  * TU shared ✘  
  * TU store ✘  
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report build-ui#9461.0 ✘  
* Compile
  * Build UI ✘  
  * TU All -   
  * TU model ✘  
  * TU service ✔ 
  * TU shared ✘  
  * TU store ✘  
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report build-ui#9496.0 ✘  
* Compile
  * Build UI ✔ 
  * TU All -   
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✘  
  * TU store ✘  
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#9496.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔ 
* Build other os/arch
  * cdsctl freebsd windows amd64 ✔ 
  * cdsctl openbsd amd64 ✔ 
  * Doc Build -   
  * Engine Build darwin amd64 ✔ 
  * Engine Build freebsd amd64 ✔ 
  * Engine Build linux arm/arm64 ✔ 
  * Engine Build openbsd amd64 ✔ 
  * Engine Build windows amd64 ✔ 
  * Plugins darwin amd64 ✔ 
  * Plugins freebsd ✔ 
  * Plugins integrations K8S -   
  * Plugins linux amd64 ✘  
  * Plugins linux arm ✔ 
  * Plugins linux arm64 ✔ 
  * Plugins openbsd ✔ 
  * Plugins windows amd64 ✔ 
  * Worker darwin / windows / freebsd / openbsd ✔ 
  * Worker linux arm ✔

**Comment by ovh-cds** [comment]
CDS Report build-ui#9502.0 ✘  
* Compile
  * Build UI ✔ 
  * TU All -   
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✘  
  * TU store ✘  
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report build-ui#9502.0 ✘  
* Compile
  * Build UI ✔ 
  * TU All -   
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✘  
  * TU store ✘  
  * TU views ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#9502.0 ✘  
* Stage 1
  * Engine Test with DB ✘

File: ui/src/app/shared/workflow/wnode/hook/hook.component.ts [hunk 1]
```diff
@@ -36,18 +45,24 @@ export class WorkflowNodeHookComponent implements OnInit {
...
+        private _store: Store, private _toast: ToastService, private _translate: TranslateService
+    ) {
+    }
 
     ngOnInit(): void {
-        this.subSelect = this._workflowEventStore.selectedHook().subscribe(h => {
-            if (this.hook && h) {
-                this.isSelected = h.uuid === this.hook.uuid;
-                return;
+        this._store.select(WorkflowState.getCurrent()).subscribe((s: WorkflowStateModel) => {
```
Comment (bnjjj) [inline ui/src/app/shared/workflow/wnode/hook/hook.component.ts]:
```suggestion
        this.subSelect = this._store.select(WorkflowState.getCurrent()).subscribe((s: WorkflowStateModel) => {
```

File: ui/src/app/shared/workflow/wnode/hook/hook.component.ts [hunk 2]
```diff
@@ -57,26 +72,37 @@ export class WorkflowNodeHookComponent implements OnInit {
...
+    receivedEvent(e: string): void {
+        switch (e) {
+            case 'details':
+                this._store.dispatch(new SelectHook({hook: this.hook, node: this.node}));
+                break;
+            case 'edit':
+                this._store.dispatch(new OpenEditModal({
+                    node: this.node,
+                    hook: this.hook
+                })).subscribe(() => {});
```
Comment (bnjjj) [inline ui/src/app/shared/workflow/wnode/hook/hook.component.ts]:
```suggestion
                }));
```

File: ui/src/app/shared/workflow/wnode/wnode.component.ts [hunk 3]
```diff
@@ -20,93 +40,82 @@ export class WorkflowWNodeComponent implements OnInit {
...
+    }
 
-        this.subSelectedNode = this._workflowEventStore.selectedNode()
-            .subscribe(n => {
-                if (n) {
-                    this.isSelected = (n.id === this.node.id);
-                } else {
-                    this.isSelected = false;
+    ngOnInit(): void {
+        this._store.select(WorkflowState.getCurrent()).subscribe((s: WorkflowStateModel) => {
```
Comment (bnjjj) [inline ui/src/app/shared/workflow/wnode/wnode.component.ts]: must save the subscriber to autounsubscribe

File: ui/src/app/shared/workflow/wnode/wnode.component.ts [hunk 4]
```diff
@@ -137,6 +146,46 @@ export class WorkflowWNodeComponent implements OnInit {
...
+                this.openTriggerModal('pipeline', false);
+                break;
+            case 'parent':
+                this.openTriggerModal('pipeline', false);
+                break;
+            case 'edit':
+                this._store.dispatch(new OpenEditModal({
+                    node: this.node,
+                    hook: null
+                })).subscribe(() => {});
```
Comment (bnjjj) [inline ui/src/app/shared/workflow/wnode/wnode.component.ts]:
```suggestion
                }));
```

File: ui/src/assets/i18n/fr.json [hunk 5]
```diff
@@ -977,12 +979,17 @@
   "workflow_name_error": "Le nom du workflow doit suivre ce pattern ^[a-zA-Z0-9._-]{1,}$",
   "workflow_name_error_duplicate": "Le nom de ce workflow existe déjà dans votre projet",
   "workflow_preview_mode": "Votre workflow est dans un état de prévisualisation",
+  "workflow_node_context_label": "Context d'éxécution",
```
Comment (bnjjj) [inline ui/src/assets/i18n/fr.json]:
```suggestion
  "workflow_node_context_label": "Contexte d'exécution",
```

File: ui/src/assets/i18n/fr.json [hunk 6]
```diff
@@ -1059,6 +1070,7 @@
   "workflow_root_context_integration": "Intégration (facultatif)",
   "workflow_root_context_mutex": "Limiter à une execution à la fois",
   "workflow_root_context_pipeline": "Pipeline",
+  "workflow_root_context_application_send_vcs_status": "Désactiver les status VCS",
```
Comment (bnjjj) [inline ui/src/assets/i18n/fr.json]:
```suggestion
  "workflow_root_context_application_send_vcs_status": "Désactiver les statuts VCS",
```

**Comment by ovh-cds** [comment]
CDS Report build-ui#9509.0 ✘  
* Compile
  * Build UI ✔ 
  * TU All -   
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✔ 
  * TU store ✘  
  * TU views ✔

File: engine/api/workflow/dao_run.go [hunk 7]
```diff
@@ -42,6 +42,7 @@ type LoadRunOptions struct {
 	WithLightTests          bool
 	WithVulnerabilities     bool
 	DisableDetailledNodeRun bool
+	Langage                 string
```
Comment (bnjjj) [inline engine/api/workflow/dao_run.go]:
```suggestion
	Language                 string
```

File: engine/api/workflow/dao_run.go [hunk 8]
```diff
@@ -928,7 +928,11 @@ func syncNodeRuns(db gorp.SqlExecutor, wr *sdk.WorkflowRun, loadOpts LoadRunOpti
 			}
 			wnr.Coverage = cov
 		}
-
+		var l = loadOpts.Langage
```
Comment (bnjjj) [inline engine/api/workflow/dao_run.go]:
```suggestion
		var l = loadOpts.Language
```

File: engine/api/workflow_run.go [hunk 9]
```diff
@@ -296,6 +296,7 @@ func (api *API) getWorkflowRunHandler() service.Handler {
 				WithArtifacts:           true,
 				WithLightTests:          true,
 				DisableDetailledNodeRun: getService(ctx) == nil,
+				Langage:                 r.Header.Get("Accept-Language"),
```
Comment (bnjjj) [inline engine/api/workflow_run.go]:
```suggestion
				Language:                 r.Header.Get("Accept-Language"),
```

File: ui/package.json [hunk 10]
```diff
@@ -59,6 +59,7 @@
     "ansi_up": "3.0.0",
     "codemirror": "5.43.0",
     "core-js": "2.6.3",
+    "cypress": "^3.2.0",
```
Comment (richardlt) [inline ui/package.json]: should be a dev dependencie ?

File: ui/src/assets/i18n/fr.json [hunk 11]
```diff
@@ -942,12 +942,14 @@
   "worker_model_username": "Nom d'utilisateur",
   "worker_model_password": "Mot de passe",
 
-
+  "workflow_modal_change_view_confirm": "Êtes-vous sure de vouloir perdre vos changements ?",
```
Comment (yesnault) [inline ui/src/assets/i18n/fr.json]:
```suggestion
  "workflow_modal_change_view_confirm": "Êtes-vous sûr de vouloir perdre vos changements ?",
```

File: ui/src/assets/i18n/fr.json [hunk 12]
```diff
@@ -982,12 +984,17 @@
...
+  "workflow_node_context_label": "Contexte d'exécution",
   "workflow_node_context_edit": "Contexte du pipeline",
   "workflow_node_context_show": "Contexte du pipeline",
+  "workflow_node_input": "Paramètres de lancement",
   "workflow_node_launch": "Actions sur le pipeline",
   "workflow_node_context_pipeline_parameter": "Paramètres du pipeline",
   "workflow_node_context_payload": "Payload par défaut",
   "workflow_node_context_payload_read_only": "Payload utilisé (lecture seule)",
+  "workflow_node_menu_trigger": "Déclencher",
+  "workflow_node_menu_edit": "Editer",
```
Comment (yesnault) [inline ui/src/assets/i18n/fr.json]:
```suggestion
  "workflow_node_menu_edit": "Éditer",
```

**Pull request merged**